src/s3/checkpoint.c

```diff
@@ -806,6 +838,36 @@ accumulate_small_file(S3BackupState *state, const char *path, int size)
 {
 	int			sizeRequired = 3 * sizeof(int) + strlen(path) + 1 + size;
 	S3TaskLocation location = 0;
+	Pointer		data;
+	uint64		dataSize;
+
+	/* First check if the file changed */
+	data = read_file(path, &dataSize);
+
+	if (data != NULL)
+	{
+		S3FileHash *entry;
+
+		if (state->hashState->pgFilesLen == state->hashState->pgFilesSize)
```

Comment:
Propose rename: pgFilesSize -> bufferSize, pgFilesLen -> currentEntry (currentSize?)

Reply:
I think that they can be named similar to `StringInfoData`:
https://github.com/postgres/postgres/blob/REL_16_STABLE/src/include/lib/stringinfo.h#L40
i.e. in the case `pgFilesSize` might become `pgFilesMaxLen`.